TrackletAlgorithm/TrackMerger.cc

```diff
@@ -0,0 +1,49 @@
+#include "TrackMerger.h"
+
+void TrackMerger(const BXType bx, TrackFit::TrackWord trackWord [kMaxProc], TrackFit::BarrelStubWord barrelStubWords[4][kMaxProc], 
+TrackFit::DiskStubWord diskStubWords[4][kMaxProc]){
+
+    #pragma HLS array_partition variable=barrelStubWords complete dim=1
+    #pragma HLS array_partition variable=diskStubWords complete dim=1
+    #pragma HLS stream variable=trackWord depth=1 dim=1
+    #pragma HLS stream variable=barrelStubWords depth=1 dim=2
+    #pragma HLS stream variable=diskStubWords depth=1 dim=2
+
+    // Comparison module
+    // count stubs in common, looking at stubs with smallest phi residual in each layer
+    for (unsigned int i = 0; i < kMaxProc; i++){
+        TrackFit trkFit;
+        trkFit.setTrackWord(trackWord[i]);
+        auto trackStubMap = trkFit.getHitMap();
+        std::cout << "Track Stub Map: " << trackStubMap << std::endl;
+        //every 3 bits is a new layer/disk - getting number of stubs in each layer
+        for (int layer = 0; layer < trackStubMap.length()/kTFHitCountSize; layer++){
+            int stubNumber = trackStubMap.range(trackStubMap.length() - kTFHitCountSize*layer, trackStubMap.length() - kTFHitCountSize*(layer + 1))
+             
+             for (unsigned int j = 0; j < kNBarrelStubs - 1; j++;){
+                trkFit.setBarrelStubWord<j>(barrelStubWords[j][i]);
```

Comment:
Template parameters always have to be determined at compile-time, and gcc isn't smart enough to unroll this loop automatically. So you'll have to do something like the following in the loop body:
```c++
switch (j) {
  case 0:
    trkFit.setBarrelStubWord<0>(barrelStubWords[0][i]);
    break;
  case 1:
    trkFit.setBarrelStubWord<1>(barrelStubWords[1][i]);
    break;
…
}
```

Something similar applies to the loop over disk stubs.

Reply:
This looks good now for the loop over barrel stubs, but for the loop over disk stubs, the template parameter needs to run from 4 to 7.